platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/reconnect/MerkleBenchmarkUtils.java

```diff
@@ -0,0 +1,185 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.benchmark.reconnect;
+
+import static com.swirlds.common.merkle.copy.MerkleInitialize.initializeTreeAfterCopy;
+import static com.swirlds.common.threading.manager.AdHocThreadManager.getStaticThreadManager;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.merkle.MerkleInternal;
+import com.swirlds.common.merkle.MerkleLeaf;
+import com.swirlds.common.merkle.MerkleNode;
+import com.swirlds.common.merkle.crypto.MerkleCryptoFactory;
+import com.swirlds.common.merkle.synchronization.LearningSynchronizer;
+import com.swirlds.common.merkle.synchronization.TeachingSynchronizer;
+import com.swirlds.common.merkle.synchronization.config.ReconnectConfig;
+import com.swirlds.common.merkle.synchronization.utility.MerkleSynchronizationException;
+import com.swirlds.common.threading.pool.StandardWorkGroup;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.platform.network.SocketConfig;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+/**
+ * A utility class to support benchmarks for reconnect.
+ */
+public class MerkleBenchmarkUtils {
+    /**
+     * Returns the following tree:
+     *
+     * <pre>
+     *             root
+     *           / |   \
+     *          A  i0  i1
+     *             /\  /\
+     *            B C D null
+     * </pre>
+     */
+    static BenchmarkMerkleInternal buildLessSimpleTree() {
+        final BenchmarkMerkleInternal root = new BenchmarkMerkleInternal("root");
+
+        final MerkleLeaf A = new BenchmarkMerkleLeaf("A");
+        final MerkleInternal i0 = new BenchmarkMerkleInternal("i0");
+        final MerkleInternal i1 = new BenchmarkMerkleInternal("i1");
+        root.setChild(0, A);
+        root.setChild(1, i0);
+        root.setChild(2, i1);
+
+        final MerkleLeaf B = new BenchmarkMerkleLeaf("B");
+        final MerkleLeaf C = new BenchmarkMerkleLeaf("C");
+        i0.setChild(0, B);
+        i0.setChild(1, C);
+
+        final MerkleLeaf D = new BenchmarkMerkleLeaf("D");
+        i1.setChild(0, D);
+        i1.setChild(1, null);
+
+        initializeTreeAfterCopy(root);
+        return root;
+    }
+
+    static <T extends MerkleNode> T hashAndTestSynchronization(
+            final MerkleNode startingTree,
+            final MerkleNode desiredTree,
+            final Configuration configuration,
+            final ReconnectConfig reconnectConfig)
```

Comment:
I may be wrong, but reconnectConfig is redundant here. It can be obtained from configuration using
```
final ReconnectConfig reconnectConfig = configuration.getConfigData(ReconnectConfig.class);
```